### PR TITLE
feat(main-app): change main app to esbuild

### PR DESCRIPTION
**Pull request by hyrious**
#1546 has been outdated too long, so I create a new PR.

Close: #1546

### CODE REVIEW

File: desktop/main-app/scripts/esbuild/paths.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import path from "path";
+
+const resolvePath = (...relativePath: string[]): string =>
+    path.resolve(__dirname, "..", "..", ...relativePath);
+
+export const dist = resolvePath("dist/index.js");
+export const preloadDist = resolvePath("dist/preload.js");
```
Comment (BlackHole1) [inline desktop/main-app/scripts/esbuild/paths.ts]:
```suggestion
export const dist = resolvePath("dist", "index.js");
export const preloadDist = resolvePath("dist", "preload.js");
```

File: desktop/main-app/scripts/esbuild/esbuild.prod.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    target: "chrome89",
+    external: [...external, "electron", "os", "path"],
+    outfile: paths.preloadDist,
+});
+
+const buildMain = esbuild.build({
+    entryPoints: [paths.entryFile],
+    bundle: true,
+    platform: "node",
+    target: "node14",
```
Comment (BlackHole1) [inline desktop/main-app/scripts/esbuild/esbuild.prod.ts:19]:
```suggestion
    target: "node16",
```

**Review by BlackHole1 — DISMISSED** [review]

File: desktop/main-app/scripts/esbuild/esbuild.prod.ts (hunk 2 above)
Comment (BlackHole1) [reply desktop/main-app/scripts/esbuild/esbuild.prod.ts:19]: wontfix

**Review by BlackHole1 — DISMISSED** [review]

**Comment by BlackHole1** [comment]
Please apply this diff.

```diff
diff --git a/desktop/main-app/scripts/esbuild/esbuild.prod.ts b/desktop/main-app/scripts/esbuild/esbuild.prod.ts
index 16669082..30aa44ce 100644
--- a/desktop/main-app/scripts/esbuild/esbuild.prod.ts
+++ b/desktop/main-app/scripts/esbuild/esbuild.prod.ts
@@ -1,5 +1,5 @@
 import esbuild from "esbuild";
-import { dotenvPlugin, external, replaceMetaPlugin } from "./esbuild.common";
+import { dotenvPlugin, external } from "./esbuild.common";
 import * as paths from "./paths";
 
 const buildPreload = esbuild.build({
@@ -9,6 +9,7 @@ const buildPreload = esbuild.build({
     target: "chrome89",
     external: [...external, "electron", "os", "path"],
     outfile: paths.preloadDist,
+    plugins: [dotenvPlugin],
 });
 
 const buildMain = esbuild.build({
@@ -19,7 +20,7 @@ const buildMain = esbuild.build({
     external: [...external, "electron", "electron-devtools-vendor"],
     sourcemap: true,
     outfile: paths.dist,
-    plugins: [dotenvPlugin, replaceMetaPlugin],
+    plugins: [dotenvPlugin],
 });
 
 Promise.all([buildPreload, buildMain]).catch(() => process.exit(1));
diff --git a/desktop/main-app/scripts/esbuild/paths.ts b/desktop/main-app/scripts/esbuild/paths.ts
index 34677b49..3d728c02 100644
--- a/desktop/main-app/scripts/esbuild/paths.ts
+++ b/desktop/main-app/scripts/esbuild/paths.ts
@@ -3,7 +3,7 @@ import path from "path";
 const resolvePath = (...relativePath: string[]): string =>
     path.resolve(__dirname, "..", "..", ...relativePath);
 
-export const dist = resolvePath("dist", "index.js");
+export const dist = resolvePath("dist", "main.js");
 export const preloadDist = resolvePath("dist", "preload.js");
 export const entryFile = resolvePath("src", "index.ts");
 export const preloadPath = resolvePath("src", "preload.ts");
```

Need rebase main branch

**Review by ylduang — APPROVED** [review]

**Pull request merged**